cmd/kube-rbac-proxy/app/options/proxyoptions.go

```diff
@@ -57,6 +64,7 @@ type ProxyOptions struct {
 func (o *ProxyOptions) AddFlags(flagset *pflag.FlagSet) {
 	flagset.StringVar(&o.Upstream, "upstream", "", "The upstream URL to proxy to once requests have successfully been authenticated and authorized.")
 	flagset.BoolVar(&o.UpstreamForceH2C, "upstream-force-h2c", false, "Force h2c to communicate with the upstream. This is required when the upstream speaks h2c(http/2 cleartext - insecure variant of http/2) only. For example, go-grpc server in the insecure mode, such as helm's tiller w/o TLS, speaks h2c only")
+	flagset.IntVar(&o.UpstreamDNSTimeout, "upstream-dns-timeout", 5, "The timeout in seconds for DNS lookups of the upstream. If set to 0, no timeout is set.")
```

Comment:
People don't need to be able to configure this. Besides, it's not actually used in the described way, there's only one spot that uses it like that.

Reply:
Since it's used only in validation, it's not really an upstream DNS timeout, as DNS lookups will be happening with all the "DNS cache miss" requests upstream, which are nowhere affected by this option.

If you feel like 5 seconds is too little, make it 10. However, if a DNS lookup is taking longer than 5s then there's something from with your infra.